.changeset/node-18.md

```diff
@@ -0,0 +1,20 @@
+---
+'@sap-cloud-sdk/connectivity': major
+'@sap-cloud-sdk/http-client': major
+'@sap-cloud-sdk/mail-client': major
+'@sap-cloud-sdk/openapi': major
+'@sap-cloud-sdk/odata-common': major
+'@sap-cloud-sdk/odata-v2': major
+'@sap-cloud-sdk/odata-v4': major
+'@sap-cloud-sdk/eslint-config': major
+'@sap-cloud-sdk/generator': major
+'@sap-cloud-sdk/test-util': major
+'@sap-cloud-sdk/util': major
+'@sap-cloud-sdk/openapi-generator': major
+'@sap-cloud-sdk/generator-common': major
+'@sap-cloud-sdk/temporal-de-serializers': major
+'@sap-cloud-sdk/resilience': major
+---
+
+
+- [Compatibility Note] Switch the compilerOptions.module to `Node16` instead of `CommonJS` for better interoperability with ES modules.([handbook](https://www.typescriptlang.org/docs/handbook/esm-node.html))
```

Comment:
```suggestion
- [Compatibility Note] Switch the compilerOptions.module to `Node16` instead of `CommonJS` for better interoperability with ES modules. See the [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/esm-node.html) for technical details if needed.
```